### PR TITLE
Lab 1 : Fibonnaci series implementation

### PR DESCRIPTION
**Pull request by NimishMudhale**
#30 Lab1 - Fibonnaci Series

### CODE REVIEW

File: 0_Fib/fib.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func Fibonacci(n int) []int {
+
+	fib := []int{}
+	t1 := 0
+	t2 := 1
```
Comment (anzdaddy) [inline 0_Fib/fib.go]: You can do a one-liner: `t1, t2 := 0, 1`.

File: 0_Fib/fib.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	sum := 0
+	for i := 0; i <= n; i++ {
+		// add the next member
+		fib = append(fib, t1)
+		sum = t1 + t2
+		t1 = t2
+		t2 = sum
```
Comment (anzdaddy) [inline 0_Fib/fib.go]: Try a one-liner here two (t1 and t1 and possibly even remove `sum` and use `t1 + t2` directly).

File: 0_Fib/fib.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func Fibonacci(n int) []int {
```
Comment (anzdaddy) [inline 0_Fib/fib.go]: Returning the slice. Nice! 👍

File: 0_Fib/fib_test.go [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+	// Given
+	r := require.New(t)
+	var buf bytes.Buffer
+	out = &buf
+
+	// When
+	main()
+
+	// Then
+	expected := strconv.Quote("Fibonnaci Series of 8:[0 1 1 2 3 5 8 13 21]")
```
Comment (anzdaddy) [inline 0_Fib/fib_test.go]: That's nine entries.

**Comment by anzdaddy** [comment]
Your code isn't auto-formatted. Do you have VSCode and the Go extension installed?

File: 0_Fib/fib.go (hunk 1 above)
Comment (NimishMudhale) [reply 0_Fib/fib.go]: done

File: 0_Fib/fib.go (hunk 2 above)
Comment (NimishMudhale) [reply 0_Fib/fib.go]: done

File: 0_Fib/fib_test.go (hunk 4 above)
Comment (NimishMudhale) [reply 0_Fib/fib_test.go]: done

File: 0_Fib/fib.go [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+	}
+	return fib
+}
+func arrayToString(a []int, delim string) string {
+	return strings.Trim(strings.Replace(fmt.Sprint(a), " ", delim, -1), "[]")
+}
+
+func main() {
+	fmt.Fprint(out, "Fibonnaci Series of 7: ", arrayToString(Fibonacci(7), "-"))
+
```
Comment (juliaogris) [inline 0_Fib/fib.go]: Unnecessary blank line.

File: 0_Fib/fib.go [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+		t1, t2 = t2, t1+t2
+	}
+	return fib
+}
+func arrayToString(a []int, delim string) string {
+	return strings.Trim(strings.Replace(fmt.Sprint(a), " ", delim, -1), "[]")
+}
+
+func main() {
+	fmt.Fprint(out, "Fibonnaci Series of 7: ", arrayToString(Fibonacci(7), "-"))
```
Comment (juliaogris) [inline 0_Fib/fib.go]: Nit: not the exact output request.

File: 0_Fib/fib.go (hunk 5 above)
Comment (juliaogris) [inline 0_Fib/fib.go]: Unnecessary blank line.

File: 0_Fib/fib.go [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+	fib, t1, t2 := []int{}, 1, 1
+	for i := 0; i < n; i++ {
+		// add the next member
+		fib = append(fib, t1)
+		t1, t2 = t2, t1+t2
+	}
+	return fib
+}
+func arrayToString(a []int, delim string) string {
```
Comment (juliaogris) [inline 0_Fib/fib.go]: This can be replaced by std lib https://golang.org/pkg/strings/#Join

**Comment by juliaogris** [comment]
Please update PR description.

File: 0_Fib/fib_test.go [hunk 8]
```diff
@@ -0,0 +1,23 @@
+package main
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestMainOutput(t *testing.T) {
```
Comment (juliaogris) [inline 0_Fib/fib_test.go]: Add another function `TestFibonacci` that works with slices (_unit_ tests).
Include corner cases: `n := 1 | 0  | -1 | overflow-big`

**Review by juliaogris — COMMENTED** [review]
Very nice and very close!

File: 0_Fib/fib.go (hunk 6 above)
Comment (NimishMudhale) [reply 0_Fib/fib.go]: done

File: 0_Fib/fib.go (hunk 5 above)
Comment (NimishMudhale) [reply 0_Fib/fib.go]: done

File: 0_Fib/fib.go (hunk 7 above)
Comment (NimishMudhale) [reply 0_Fib/fib.go]: done

File: 0_Fib/fib_test.go (hunk 8 above)
Comment (NimishMudhale) [reply 0_Fib/fib_test.go]: done

File: 0_Fib/fib_test.go [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+
+	// When
+	main()
+
+	// Then
+	expected := "1\n1\n2\n3\n5\n8\n13"
+	actual := buf.String()
+	r.Equalf(expected, actual, "Unexpected output in main()")
+}
+func TestFibonnaci1(t *testing.T) {
```
Comment (anzboi) [inline 0_Fib/fib_test.go]: Newline after closing `}`

File: 0_Fib/fib_test.go [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+	//Given
+	assert := assert.New(t)
+	//when
+	fibonnaciArray := Fibonacci(1)
+
+	//Then
+	assert.Equal([]int{1}, fibonnaciArray)
+
```
Comment (anzboi) [inline 0_Fib/fib_test.go]: Delete this extra line. Add newline after closing `}`

File: 0_Fib/fib_test.go [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+}
+func TestFibonnaci2(t *testing.T) {
+	//Given
+	assert := assert.New(t)
+	//when
+	fibonnaciArray := Fibonacci(-1)
+
+	//Then
+	assert.Equal([]int{}, fibonnaciArray)
+
```
Comment (anzboi) [inline 0_Fib/fib_test.go]: Delete this extra line. Add newline after closing `}`.

File: 0_Fib/fib_test.go [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+func TestFibonnaci3(t *testing.T) {
+	//Given
+	assert := assert.New(t)
+	//when
+	fibonnaciArray := Fibonacci(0)
+
+	//Then
+	assert.Equal([]int{}, fibonnaciArray)
+
+}
```
Comment (anzboi) [inline 0_Fib/fib_test.go]: Add newline at the end of file.

**Review by anzboi — CHANGES_REQUESTED** [review]
Great work! LGTM after fixing few format issues in `fib_test.go`

**Review by juliaogris — COMMENTED** [review]
- PR should be called lab2 bubble sort.
- Update PR description.
-   Click `Update branch` and locally `git pull origin master` before adding new commits.
- Issue PR from feature branch not your master.
-   Please edit your commits to follow [recommendations](https://chris.beams.io/posts/git-commit/)
    -   Capitalize the subject line
    -   Use the imperative mood in the subject line
    -   Fold fixup commits into previous ones with `git rebase -i HASH`
I can give a hand with this if you want

**Pull request closed without merging**